src/ipc_cache.c

```diff
@@ -0,0 +1,225 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <stdbool.h>
+
+#include "base_alloc_global.h"
+#include "ipc_cache.h"
+#include "utils_common.h"
+#include "utils_concurrency.h"
+#include "utils_log.h"
+#include "utlist.h"
+
+struct ipc_handle_cache_entry_t;
+
+typedef struct ipc_handle_cache_entry_t *hash_map_t;
+typedef struct ipc_handle_cache_entry_t *lru_list_t;
+
+typedef struct ipc_handle_cache_entry_t {
+    UT_hash_handle hh;
+    struct ipc_handle_cache_entry_t *next, *prev;
+    ipc_mapped_handle_cache_key_t key;
+    uint64_t ref_count;
```

Comment:
this field is never read - only set. Did you forgot about any feature?

Reply:
Yeah, as I mentioned in [here](https://github.com/oneapi-src/unified-memory-framework/pull/736#discussion_r1828160757) I am going to implement it in next PR. The `ref_count` will be needed when `max_size` is used.